gremlin-go/driver/client.go

```diff
@@ -43,6 +43,7 @@ type ClientSettings struct {
 	// Maximum number of concurrent connections. Default: number of runtime processors
 	MaximumConcurrentConnections int
 	Session                      string
+	ConnectionTimeout            time.Duration
```

Comment:
Can you please move ConnectionTimeout in the various properties and method parameters to be under WriteDeadline for better oganization?

Other than that, LGTM